app/models/user.rb

```diff
@@ -31,6 +33,16 @@ def notified_by_ride?(ride)
     notify? && notifying_rides.include?(ride)
   end
 
+  def display_name
+    # Display email if name not present
+    if name.presence
+      return name
+    end
+
+    match = email.match(/\A(.*)@/)
+    "[#{match.captures[0]}]"
+  end
```

Comment:
This can stay here for now, but really this is view code and should exist in a helper.  However, it will be useful to have this method for doing notifications until we can wework those aswell

Reply:
I think that if we do have some sort of "default human name for users", it probably makes more sense for that to live in the model than in view code as it is more a property of the object than a view-specific one. As it stands it feels to me like we just have "default human name for users" everywhere we would put a person's name, and as such I thought it would be a good idea to move it into the model, and then leave anything less general (first name, last name, capitalization changes, etc.) to view code, as I'm guessing things like that will come up.

Reply:
My worry is that we shouldn't allow processing for human readability in the model.  The email matcher is doing something like that in this function.  It's fine and useful right now because view level stuff has leaked into the controllers too, but it is a sign that we need to refactor all of the things that refer to a "human readable name" back into view code.